### PR TITLE
[HHH-17294] DeepCopy non-Embedded JSON or XML JdbcTypCode attribute using FormatMapper

### PR DESCRIPTION
**Pull request by The-Huginn**
https://hibernate.atlassian.net/browse/HHH-17294

### CODE REVIEW

**Comment by The-Huginn** [comment]
Just a thought.
Internally `Objects.equals` is used to decide, whether an attribute was changed. Thus the user should overwrite `equals` method to work correctly, as we are deep copying new instance and then upon these instances `equals` is called.

File: hibernate-core/src/test/java/org/hibernate/orm/test/mapping/basic/JsonAggregateTest.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	@JiraKey("HHH-17294")
+	public void testDirtyChecking() {
+		sessionFactoryScope().inTransaction(
+				entityManager -> {
+					JsonHolder jsonHolder = entityManager.find( JsonHolder.class, 1L );
+					assertEquals("String 'abc'", jsonHolder.getAggregate().getTheString());
+					jsonHolder.getAggregate().setTheString( "MyString" );
+					entityManager.flush();
+					entityManager.clear();
+					// Fails, when it should pass
```
Comment (ElTav) [inline hibernate-core/src/test/java/org/hibernate/orm/test/mapping/basic/JsonAggregateTest.java:63]: Rm the comment?

File: hibernate-core/src/test/java/org/hibernate/orm/test/mapping/basic/Aggregate.java [hunk 2]
```diff
@@ -0,0 +1,469 @@
...
+import java.time.ZonedDateTime;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.Objects;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@Access( AccessType.PROPERTY )
+public class Aggregate {
```
Comment (ElTav) [inline hibernate-core/src/test/java/org/hibernate/orm/test/mapping/basic/Aggregate.java:45]: I didn't for my test, but might make sense to have `JsonAggregate` extend this class to reduce duplication?

**Review by ElTav — COMMENTED** [review]
Would it be possible to hook into the bytecode dirty object tracking (a la the fix [here](https://github.com/hibernate/hibernate-orm/pull/6765/files) to avoid having to serialize + deserialize the object on every save?

**Comment by ElTav** [comment]
Hey @The-Huginn, any luck with this?

**Comment by The-Huginn** [comment]
Hi @ElTav , still working on this but it's getting quite large. Hopefully I will be able to update this PR soon

**Comment by beikov** [comment]
Thanks for your effort @The-Huginn, I took it to the next level and also implemented some `JavaType` related methods to use the `FormatMapper`. Also, I realized that it might be best to treat values for types that use the `MutableMutabilityPlan` always as dirty since this default `MutabilityPlan` actually doesn't do copying.

Superseded by https://github.com/hibernate/hibernate-orm/pull/7525

**Pull request closed without merging**